include/deal.II/base/polynomials_rt_bubbles.h

```diff
@@ -0,0 +1,189 @@
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__polynomials_rt_bubbles_h
+#define dealii__polynomials_rt_bubbles_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomials_raviart_thomas.h>
+#include <deal.II/base/table.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class implements the <i>H<sup>div</sup></i>-conforming, vector-valued
+ * enhanced Raviart-Thomas polynomials.
+ *
+ * Similarly to the classical Raviart-Thomas space, the enhanced Raviart-Thomas
+ * polynomials are constructed such that the divergence is in the tensor product
+ * polynomial space <i>Q<sub>k-1</sub></i>.
+ *
+ * This space is of the form <i>V<sub>k</sub> = RT<sub>k-1</sub> + B<sub>k</sub></i>,
+ * where <i>B<sub>k</sub></i> is defined as follows:
+ * <dl>
+ * <dt> In 2D:
+ * @f{align*}{
+ *  B_k^1(E) = \text{span}\left\{x^{a_1-1} y^{a_2}\begin{pmatrix} (a_2+1) x \\
+ *    -a_1 y \end{pmatrix}\text{ : } a_2=k \right\} \\
+ *  B_k^2(E) = \text{span}\left\{x^{b_1} y^{b_2-1}\begin{pmatrix} -b_2 x \\
+ *     (b_1+1) y \end{pmatrix}\text{ : } b_1=k \right\}
+ * @f}
+ *
+ * <dt> In 3D:
+ *  @f{align*}{
+ *   B_k^1(E) = \text{span}\left\{x^{a_1-1} y^{a_2} z^{a_3}\begin{pmatrix} (a_2+a_3+2) x \\
+ *     -a_1 y \\ -a_1 z \end{pmatrix}\text{ : } a_2=k \text{ or } a_3=k \right\},\\
+ *   B_k^2(E) = \text{span}\left\{x^{b_1} y^{b_2-1} z^{b_3}\begin{pmatrix} -b_2 x \\
+ *     (b_1+b_3+2) y \\ -b_2 z \end{pmatrix}\text{ : } b_1=k \text{ or } b_3=k \right\},\\
+ *   B_k^3(E) = \text{span}\left\{x^{c_1}y^{c_2}z^{c_3-1}\begin{pmatrix} -c_3 x \\
+ *     -c_3y \\ (c_1+c_2+2)z \end{pmatrix}\text{ : } c_1=k \text{ or } c_2=k \right\},
+ *  @f}
+ * </dl>
+ * where $0 \le a_1, a_2, a_3 \le k$.
+ *
+ * @note Unlike the classical Raviart-Thomas space, the lowest order for the enhaced space
+ * is 1, similarly to the Brezzi-Douglas-Marini (BDM) polynomial space.
+ *
+ * The total dimension of the space <i>dim(V<sub>k</sub>) = d*(k+1)^d</i>, where <i>d</i> is the
+ * space dimension. This allows to associate shape functions with the Gauss-Lobatto quadrature
+ * points as shown in the figures below.
+ *
+ * <table> <tr> <td align="center">
+ * @image html http://www.dealii.org/images/shape-functions/RTBubbles/rtbubbles.png
```

Comment:
actually, just put the picture into `doc/doxygen/images`, git add it to the patch, and then change this line to
```
  @image html rtbubbles.png
```
doxygen will automatically find it in the directory mentioned above.